### PR TITLE
update setup wizard and main ui to accommodate https

### PR DESCRIPTION
**Pull request by elvece**

### CODE REVIEW

**Comment by elvece** [comment]
Didn't add anything to final page of setup wizard. Felt the prompt to download and launch was sufficient with the updates I made to the download page, which now looks like:
Screenshot 2023-07-14 at 2 45 01 PM

**Comment by MattDHill** [comment]
Are the addresses not showing on purpose?

File: frontend/projects/ui/src/app/pages/apps-routes/app-interfaces/app-interfaces.page.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ export class AppInterfacesPage {
             ? 'https://' + uiAddresses['lan-address']
             : '',
           'tor-address': uiAddresses['tor-address']
-            ? 'http://' + uiAddresses['tor-address']
+            ? 'https://' + uiAddresses['tor-address']
```
Comment (dr-bonez) [inline frontend/projects/ui/src/app/pages/apps-routes/app-interfaces/app-interfaces.page.ts:55]: not for service uis

File: frontend/projects/ui/src/app/services/api/embassy-live-api.service.ts [hunk 2]
```diff
@@ -449,10 +449,9 @@ export class LiveApiService extends ApiService {
 
   private openWebsocket<T>(config: WebSocketSubjectConfig<T>): Observable<T> {
     const { location } = this.document.defaultView!
-    const protocol = location.protocol === 'http:' ? 'ws' : 'wss'
     const host = location.host
 
-    config.url = `${protocol}://${host}/ws${config.url}`
+    config.url = `wss://${host}/ws${config.url}`
```
Comment (dr-bonez) [inline frontend/projects/ui/src/app/services/api/embassy-live-api.service.ts:454]: we still need to support ws on lan

**Review by dr-bonez — CHANGES_REQUESTED** [review]

**Comment by elvece** [comment]
> Are the addresses not showing on purpose?

yes because i was just opening the raw html page in the browser

File: frontend/projects/setup-wizard/src/app/pages/success/success.page.ts [hunk 3]
```diff
@@ -49,7 +49,7 @@ export class SuccessPage {
       const ret = await this.api.complete()
       if (!this.isKiosk) {
         this.torAddress = ret['tor-address']
-        this.lanAddress = ret['lan-address'].replace('https', 'http')
```
Comment (elvece) [inline frontend/projects/setup-wizard/src/app/pages/success/success.page.ts:52]: do we actually want to leave this as is?

Comment (dr-bonez) [reply frontend/projects/setup-wizard/src/app/pages/success/success.page.ts:52]: I'm not actually sure what this is for

Comment (dr-bonez) [reply frontend/projects/setup-wizard/src/app/pages/success/success.page.ts:52]: oh, actually, I understand. Yes, we want to leave it, @MattDHill  still wants to encourage http immediately after setup on lan

File: frontend/projects/ui/src/app/app.component.ts [hunk 4]
```diff
@@ -38,7 +38,15 @@ export class AppComponent implements OnDestroy {
...
-  ngOnInit() {
+  async ngOnInit() {
+    if (location.protocol === 'http:') {
+      // see if site is available securely
+      const res = await fetch(window.location.href.replace('http', 'https'))
+      if (res && res.status === 200) {
+        // redirect
+        window.location.protocol = 'https:'
+      }
+    }
```
Comment (elvece) [inline frontend/projects/ui/src/app/app.component.ts:49]: redirect @dr-bonez 
wont work for localhost, but tested setting `window.location.protocol = 'https:'` when on http of a current live server and it redirects properly

**Review by dr-bonez — APPROVED** [review]

**Pull request merged**